### PR TITLE
python3Packages.sphinxHook: Fix cross compilation

### PR DESCRIPTION
**Pull request by kip93**
Fixes build of wrapt. Also removed the now irrelevant workaround on beautifulsoup4.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (cross compiled from x86_64-linux)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kip93** [comment]
Changed target branch to staging and rebased due to number of rebuilds.

**Review by FRidh — APPROVED** [review]

**Comment by Artturin** [comment]
PR no longer necessary since https://github.com/NixOS/nixpkgs/pull/247245

**Comment by kip93** [comment]
Just rebased with staging to be sure, but it still fails to cross compile wrap without this, with the same error

**Review by Artturin — APPROVED** [review]
Sorry I forgot that my pr only fixed splicing `python3Packages.callPackage { python }:` while the hook splicing within the set (used by callPackage) is still broken and will eventually be fixed by https://github.com/NixOS/nixpkgs/pull/228139

the rebuilds are a bit weird so i'll check why that is happening

File: pkgs/development/interpreters/python/hooks/default.nix [hunk 1]
```diff
@@ -221,9 +221,9 @@ in {
     inherit (pkgs.buildPackages) makeWrapper;
   };
 
-  sphinxHook = callPackage ({ makePythonHook, sphinx, installShellFiles }:
+  sphinxHook = callPackage ({ makePythonHook, buildPackages, installShellFiles }:
     makePythonHook {
       name = "python${python.pythonVersion}-sphinx-hook";
-      propagatedBuildInputs = [ sphinx installShellFiles ];
+      propagatedBuildInputs = [ buildPackages.sphinx installShellFiles ];
```
Comment (Artturin) [inline pkgs/development/interpreters/python/hooks/default.nix:227]:
```
$ nixDiff "python311Packages.beautifulsoup4"
- /nix/store/ppf8gh8ync78x5h4yms6qmchmmwk7wci-python3.11-beautifulsoup4-4.12.2.drv:{out}
+ /nix/store/iyjp684sdhgwb1g6qbfbqjfrl9jgmi25-python3.11-beautifulsoup4-4.12.2.drv:{out}
• The input derivation named `python3.11-sphinx-hook` differs
  - /nix/store/0lwm7xy371z33qksx36kvipcra95r5p3-python3.11-sphinx-hook.drv:{out}
  + /nix/store/q7vljpzdxcrsm82in7c6ya7jxshk1cf2-python3.11-sphinx-hook.drv:{out}
  • The set of input derivation names do not match:
      - python3.11-sphinx-5.3.0
      + python3.10-sphinx-5.3.0
  • The environments do not match:
      propagatedBuildInputs=''
      /nix/store/36dwajhn8fl6n5mam2kdqx6fx2fibqiw-python3.11-sphinx-5.3.0/nix/store/0rfrksr3p3vxl9calc003x7l1ka9qpkh-python3.10-sphinx-5.3.0 /nix/store/bv95azmp1g6ksp2wl35kxkjlbqj48hz3-install-shell-files
  ''
• Skipping environment comparison
```

```suggestion
      propagatedBuildInputs = [ pythonForBuild.sphinx installShellFiles ];
```

**Comment by Artturin** [comment]
I checked and there's really no rebuilds, rebuild count is likely just because of a rebase gone wrong

**Pull request merged**